### PR TITLE
move access request logic to keycloak controller/service. Update to c…

### PR DESCRIPTION
**Pull request by devinleighsmith**
…all keycloak update user method.

### CODE REVIEW

File: backend/dal.keycloak/Partials/PimsKeycloakUserService.cs [hunk 1]
```diff
@@ -160,6 +162,51 @@ public async Task<Entity.User> UpdateUserAsync(Entity.User user)
...
             return euser;
         }
+
+        /// <summary>
+        /// Updates the specified access request in the datasource. if the request is granted, update the associated user as well.
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        /// <returns></returns>
+        public async Task<Entity.AccessRequest> UpdateAccessRequest(Entity.AccessRequest entity)
```
Comment (Fosol) [inline backend/dal.keycloak/Partials/PimsKeycloakUserService.cs]: Rename to `UpdateAccessRequestAsync` to keep in alignment with other async pattern.

File: backend/api/Areas/Keycloak/Controllers/UserController.cs [hunk 2]
```diff
@@ -107,6 +107,20 @@ public async Task<IActionResult> UpdateUserAsync(Guid id, [FromBody] Model.Updat
 
             return new JsonResult(result);
         }
+
+        /// <summary>
+        /// Update an access request, generally to grant/deny it.
+        /// </summary>
+        /// <returns></returns>
+        [HttpPut("access/request")]
+        public async Task<IActionResult> UpdateAccessRequests(Pims.Api.Models.AccessRequestModel updateModel)
```
Comment (Fosol) [inline backend/api/Areas/Keycloak/Controllers/UserController.cs]: Rename to `UpdateAccessRequestsAsync` to keep in alignment with other async naming pattern.

File: backend/dal/Services/Admin/Concrete/UserService.cs [hunk 3]
```diff
@@ -151,59 +151,36 @@ public override void Remove(User entity)
...
-            this.Context.Entry(accessRequest).CurrentValues.SetValues(entity);
-            accessRequest.UpdatedById = this.User.GetUserId();
-            accessRequest.UpdatedOn = DateTime.UtcNow;
-            this.Context.AccessRequests.Update(accessRequest);
-
-            this.Context.CommitTransaction();
-            return accessRequest;
+                .AsNoTracking()
+                .Where(ar => ar.Id == id)
+                .SingleOrDefault();
```
Comment (Fosol) [inline backend/dal/Services/Admin/Concrete/UserService.cs]: Change to `FirstOrDefault()`, although I'm not certain it's a performance gain in SQL.

Comment (Fosol) [reply backend/dal/Services/Admin/Concrete/UserService.cs]: I suspect you should also add `?? throw KeyNotFoundException();` as well.

File: backend/dal.keycloak/Partials/PimsKeycloakUserService.cs [hunk 4]
```diff
@@ -160,6 +162,51 @@ public async Task<Entity.User> UpdateUserAsync(Entity.User user)
...
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        /// <returns></returns>
+        public async Task<Entity.AccessRequest> UpdateAccessRequestAsync(Entity.AccessRequest entity)
+        {
+            entity.ThrowIfNull(nameof(entity));
+            entity.ThrowIfNull(nameof(entity.UserId));
+
+            this._user.ThrowIfNotAuthorized(Permissions.AdminUsers, Permissions.AgencyAdmin);
+            var accessRequest = _pimsAdminService.User.GetAccessRequestNoTracking(entity.Id);
+            if (!accessRequest.IsGranted != true && entity.IsGranted == true)
```
Comment (asanchezr) [inline backend/dal.keycloak/Partials/PimsKeycloakUserService.cs]: /nit: Please clean up the logic here. Double negations are harder to understand.

Isn't `if (!accessRequest.IsGranted != true && entity.IsGranted == true)` the same as `if (accessRequest.IsGranted == true && entity.IsGranted == true)` ?

If the property can be null I'd rather do `var value = accessRequest.IsGranted ?? false` and then check the value?

Comment (devinleighsmith) [reply backend/dal.keycloak/Partials/PimsKeycloakUserService.cs]: should actually be accessRequest.IsGranted != true, not sure how this works as is. Will update in my next PR.

**Pull request merged**